### PR TITLE
fix(issue 217): validate calendar renaming on input blur

### PR DESCRIPTION
**Pull request by sylzys**
Addressing issue #217

### CODE REVIEW

File: client/app/views/menu_item.coffee [hunk 1]
```diff
@@ -84,15 +92,15 @@ module.exports = class MenuItemView extends BaseView
         input.focus()
         input[0].setSelectionRange 0, calendarName.length
 
-        # Binds event to actually rename the calendar (enter key)
+        # Binds event to actually rename the calendar (enter key and blur)
+        input.blur (event) =>
+            console.log(calendarName)
```
Comment (jsilvestre) [inline client/app/views/menu_item.coffee]: Please remove the console.log

File: client/app/views/menu_item.coffee [hunk 2]
```diff
@@ -65,6 +65,14 @@ module.exports = class MenuItemView extends BaseView
         @$('.dropdown-toggle').on 'click', @hideColorPicker
 
 
+    #setting rename validation function to avoid code duplication
+    onRenameValidation: (input, calendarName) ->
+        @startSpinner()
+        # removes the binding to prevent memory leak
+        input.off 'keyup'
```
Comment (jsilvestre) [inline client/app/views/menu_item.coffee]: You `off`  "keyup" but not "blur", is it on purpose?

**Comment by m4dz** [comment]
Thanks for your PR!

For events binding, as the `input` object is a jQuery object, I suggest we use a global listener such as 
```coffee
input.on 'blur keyup', @onRenameValidation
```

and then isolate all logics for the validation in the `@onRenameValidation` method. We also don't forget to self-detach events with an off method ;).

I also suggest to attach a second `keyup` event to input to deal with the colorHash case.

@sylzys are you ok with those suggestions? Can you update your PR to take care of them?

Thanks!

**Comment by jsilvestre** [comment]
I added more comments! Again, thank you for the your work.

**Comment by sylzys** [comment]
Moved all 3 events actions into onRenameValidation().
@m4dz what do you mean by "colorhash case" ? 
The keyup event handled 2 cases right ? esc key and return key?

In my new commit, the badge color is updated as soon as the user presses enter or blurs the field out.

**Comment by m4dz** [comment]
Well, you're right: the blur event is caught at `else if (key is 13 or event.type is 'focusout')`, so it should work. Let's go!

**Comment by sylzys** [comment]
@m4dz it's done

**Pull request merged**